method/SCAN-IT/method_scanit.py

```diff
@@ -0,0 +1,162 @@
+    help="Path to neighbor definitions. Square matrix (not necessarily symmetric) where each row contains the neighbors of this observation (as mtx).",
+    required=False,
+)
+parser.add_argument("-d", "--out_dir", help="Output directory.", required=True)
+parser.add_argument(
+    "--dim_red",
+    help="Reduced dimensionality representation (e.g. PCA).",
+    required=False,
+)
+parser.add_argument("--image", help="Path to H&E staining.", required=False)
+parser.add_argument(
+    "--n_clusters", help="Number of clusters to return.", required=True, type=int
+)
+parser.add_argument(
+    "--technology",
+    help="The technology of the dataset (Visium, ST, imaging-based).",
+    required=True,
+)
+parser.add_argument(
+    "--seed", help="Seed to use for random operations.", required=True, type=int
+)
+parser.add_argument(
+    "--config",
+    help="Optional config file (json) used to pass additional parameters.",
+    required=False,
+)
+
+args = parser.parse_args()
+
+from pathlib import Path
+
+out_dir = Path(args.out_dir)
+
+# Output files
+label_file = out_dir / "domains.tsv"
+embedding_file = out_dir / "embedding.tsv"
+
+n_clusters = args.n_clusters
+technology = args.technology
+seed = args.seed
+
+def get_anndata(args):
+    import anndata as ad
+    import numpy as np
+    import pandas as pd
+    import scipy as sp
+    from PIL import Image
+
+    observations = pd.read_table(args.observations, index_col=0)
+    features = pd.read_table(args.features, index_col=0)
+
+    coordinates = (
+        pd.read_table(args.coordinates, index_col=0)
+        .loc[observations.index, :]
+        .to_numpy()
+    )
+
+    adata = ad.AnnData(obs=observations, var=features, obsm={"spatial": coordinates})
+
+    if args.matrix is not None:
+        X = sp.io.mmread(args.matrix)
+        if sp.sparse.issparse(X):
+            X = X.tocsr()
+        adata.X = X
+
+    if args.neighbors is not None:
+        adata.obsp["spatial_connectivities"] = sp.io.mmread(args.neighbors).T.tocsr()
+
+    # Filter by selected samples/features
+    if "selected" in adata.obs.columns:
+        adata = adata[observations["selected"].astype(bool), :]
+    if "selected" in adata.var.columns:
+        adata = adata[:, features["selected"].astype(bool)]
+
+    if args.dim_red is not None:
+        adata.obsm["reduced_dimensions"] = (
+            pd.read_table(args.dim_red, index_col=0).loc[adata.obs_names].to_numpy()
+        )
+
+    if args.image is not None:
+        adata.uns["image"] = np.array(Image.open(args.img))
+    else:
+        adata.uns["image"] = None
+
+    return adata
+
+
+adata = get_anndata(args)
+
+# Load config file
+import json
+
+with open(args.config, "r") as f:
+    config = json.load(f)
+    
+# Set the seed
+import random
+random.seed(seed)
+
+import scanit
+import scanpy as sc
+import pandas as pd
+import warnings
+
+# Construct the spatial graph
+scanit.tl.spatial_graph(adata, method='alpha shape', alpha_n_layer=2, knn_n_neighbors=5)
```

Comment:
1. Is it possible to use existig neighbor definitions?
1. Does it make sense to test different parameters here?

Reply:
Unfortunately the developers do not provide any insights into the parameter choice, so I am not sure which values it is sensible to test. So the only thing I can do is create a config file with random combinations.